x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/RuntimeFields.java

```diff
@@ -36,13 +50,39 @@
 import org.elasticsearch.xpack.runtimefields.mapper.KeywordScriptFieldType;
 import org.elasticsearch.xpack.runtimefields.mapper.LongFieldScript;
 import org.elasticsearch.xpack.runtimefields.mapper.LongScriptFieldType;
+import org.elasticsearch.xpack.runtimefields.mapper.SimplePattern;
+import org.elasticsearch.xpack.runtimefields.mapper.SimplePattern.GrokHelper;
 import org.elasticsearch.xpack.runtimefields.mapper.StringFieldScript;
 
+import java.util.Collection;
 import java.util.List;
 import java.util.Map;
+import java.util.function.Supplier;
 
 public final class RuntimeFields extends Plugin implements MapperPlugin, ScriptPlugin, ActionPlugin {
 
+    static final Setting<TimeValue> GROK_WATCHDOG_INTERVAL = Setting.timeSetting(
+        "runtime_fields.grok.watchdog.interval",
+        TimeValue.timeValueSeconds(1),
+        Setting.Property.NodeScope
+    );
+    static final Setting<TimeValue> GROK_WATCHDOG_MAX_EXECUTION_TIME = Setting.timeSetting(
+        "runtime_fields.grok.watchdog.max_execution_time",
+        TimeValue.timeValueSeconds(1),
+        Setting.Property.NodeScope
+    );
```

Comment:
In ingest these are node level settings so I made them node level settings. That might have been an accident of history. Runtime fields *usually* run inside of a search context so we could maybe do better than the watchdog that we have, but that seems like a problem for another time.